### PR TITLE
Markdown support

### PR DESCRIPTION
**Pull request by TitusEntertainment**
Adds support for markdown. The object requested looks like the one here: https://tiptap.dev/guide/output

### CODE REVIEW

**Comment by TitusEntertainment** [comment]
Seems to come with a surprising amount of other stuff sorry about that @binaryoverload

File: src/routes/tickets/util/getMetadataFromTicket.ts [hunk 1]
```diff
@@ -1,11 +1,29 @@
 import { ITicket, ITicketMetaData } from '../controllers/ticketController';
 
+
 export const getMetadataFromTicket = (data: ITicket[]): ITicketMetaData[] => {
```
Comment (binaryoverload) [inline src/routes/tickets/util/getMetadataFromTicket.ts:4]: Why is there an array of tickets passed in if you are getting the metadata for a specific ticket

File: src/routes/tickets/util/getMetadataFromTicket.ts [hunk 2]
```diff
@@ -1,11 +1,29 @@
...
-	const metadata = data;
+  const metadata = data;
+  let flattenedMessage = '';
+
+  // Take all the messages put them together into a single string and if the string is equals or grater than 50 return. Once 50 add three dots at the end of te string and return that as previewString
```
Comment (binaryoverload) [inline src/routes/tickets/util/getMetadataFromTicket.ts:8]: The code doesn't match this comment. The code currently takes each text paragraph in the first message and flatmaps it.

File: src/routes/tickets/util/getMetadataFromTicket.ts [hunk 3]
```diff
@@ -1,11 +1,29 @@
...
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  metadata[0].messages[0].body.content.forEach(message => {
+    message.content?.forEach(content => {
+      if (flattenedMessage.length >= 50) return;
+      if (content.type === 'text' && typeof content.text === 'string') {
+        flattenedMessage = `${flattenedMessage} ${content.text}`;
```
Comment (binaryoverload) [inline src/routes/tickets/util/getMetadataFromTicket.ts:16]:
```suggestion
        flattenedMessage += " " + content.text;
```

File: src/routes/tickets/util/getMetadataFromTicket.ts [hunk 4]
```diff
@@ -1,11 +1,29 @@
...
+      }
+    });
+  });
 
-	metadata[0].previewText = metadata[0].messages[0].text.slice(0, 50);
+  const newMessageArr = flattenedMessage.slice(0, 50).split('');
+  newMessageArr.push('...');
+  metadata[0].previewText = newMessageArr.join('');
```
Comment (binaryoverload) [inline src/routes/tickets/util/getMetadataFromTicket.ts:23]: I think a simpler version might be just 
```js
metadata[0].previewText = flattenedMessage.slice(0, 50) + "..."
```

File: src/routes/tickets/controllers/ticketController.ts [hunk 5]
```diff
@@ -2,98 +2,127 @@ import { RequestHandler } from 'express';
...
+    type: string;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    attrs?: Record<string, any>;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    [key: string]: any;
+  }[];
+  text?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  [key: string]: any;
+}
```
Comment (binaryoverload) [inline src/routes/tickets/controllers/ticketController.ts:26]: What is this for? Comments pls <3

File: src/routes/tickets/controllers/replyTicket.ts [hunk 6]
```diff
@@ -1,65 +1,66 @@
...
-		},
-		{
-			returnOriginal: false,
-		},
-	);
+  const updatedData = await getDB().tickets.findOneAndUpdate(
+    { _id },
+    {
+      $set: {
+        status: TicketStatus.open,
```
Comment (binaryoverload) [inline src/routes/tickets/controllers/replyTicket.ts:26]: Out of curiosity, why the change from updated to open?

**Review by binaryoverload — CHANGES_REQUESTED** [review]

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 1 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:4]: True, don't need to do that. I'll fix

Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:4]: Because the callback object sets the "previewText" on the main object and not in the messages array

Comment (mrjvs) [reply src/routes/tickets/util/getMetadataFromTicket.ts:4]: ?!

Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:4]: Let me clarify :)

the object hat is returned by the function alters the main object from the Ticket i.e the previewText.

messages only exists on the Ticket object but the previewText will not exist in the messages object but rather the ticket object which then becomes the metadata ticket object

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 2 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:8]: Well, then rest does what's commented :)

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 3 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:16]: Any specific reason for making this change?

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 4 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:23]: @binaryoverload Can you make this a suggestion like the one above regarding the flatmap?

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 3 above)
Comment (binaryoverload) [reply src/routes/tickets/util/getMetadataFromTicket.ts:16]: It's just more clear IMO, and I think in @mrjvs's opinion too

File: src/routes/tickets/controllers/ticketController.ts (hunk 5 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/ticketController.ts:26]: Oh yeah only commented this on the FE I think. This comes from the lib tiptap JSONContent is essentially the type that is returned from editor.getJSON() on the FE.

Comment (binaryoverload) [reply src/routes/tickets/controllers/ticketController.ts:26]: `TipTapContent` -> Maybe as a better name?

File: src/routes/tickets/controllers/replyTicket.ts (hunk 6 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/replyTicket.ts:26]: Forgot to revoke this change, will do so now.

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 4 above)
Comment (binaryoverload) [inline src/routes/tickets/util/getMetadataFromTicket.ts:23]:
```suggestion
  metadata[0].previewText = flattenedMessage.slice(0, 50) + "..."
```

**Review by binaryoverload — CHANGES_REQUESTED** [review]

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 3 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:16]: Fair enough 👍

File: src/routes/tickets/controllers/ticketController.ts (hunk 5 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/ticketController.ts:26]: Maybe, but the type on the Fe is JSONContent (This type/interface is taken from their lib) so I think it makes more sense to keep it the same between the two projects/services

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 3 above)
Comment (mrjvs) [reply src/routes/tickets/util/getMetadataFromTicket.ts]: 1. its easier to read. it litterally does + " " + msg. there is no need for templating.
  - if you want you can also do `msg = msg.concat(" ", otherMsg);`
2. its so much faster

Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts]: Will go with the concat route then.

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 1 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:4]: Wait nvm

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 2 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:8]: Nvm

File: src/routes/tickets/controllers/ticketController.ts (hunk 5 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/ticketController.ts:26]: @binaryoverload

File: src/routes/tickets/util/getMetadataFromTicket.ts (hunk 1 above)
Comment (TitusEntertainment) [reply src/routes/tickets/util/getMetadataFromTicket.ts:7]: My new solution worked first try 🥳 
external-content duckduckgo com

File: src/routes/tickets/controllers/ticketController.ts (hunk 5 above)
Comment (binaryoverload) [reply src/routes/tickets/controllers/ticketController.ts:26]: I disagree personally, I would name it what it makes sense to. Even when matching with the FE, a different name would offer much more clarity.

If that's still a no, then please at least add a comment.

@mrjvs opinion?

Comment (mrjvs) [reply src/routes/tickets/controllers/ticketController.ts:26]: agree with bin bin

Comment (TitusEntertainment) [reply src/routes/tickets/controllers/ticketController.ts:26]: Alright, the council has spoken. 👨‍⚖️

File: src/routes/tickets/controllers/getUnassignedTickets.ts [hunk 7]
```diff
@@ -23,7 +23,7 @@ const getUnassignedTickets = async (req: Request, res: Response) => {
   const tickets: IPaginateResult<ITicket> = await find(getDB().tickets, {
     limit: parseInt(process.env.PAGINATION_LIMIT, 10),
     query: {
-      status: TicketStatus.updated,
+      status: TicketStatus.open,
```
Comment (binaryoverload) [inline src/routes/tickets/controllers/getUnassignedTickets.ts:26]: Is this meant to be open? I noticed you said you would revert this elsewhere

File: src/routes/tickets/controllers/getUserAuthoredTickets.ts [hunk 8]
```diff
@@ -15,7 +15,17 @@ const getUserAuthoredTickets = async (req, res) => {
 
   const tickets = await find(getDB().tickets, {
     limit: parseInt(process.env.PAGINATION_LIMIT, 10),
-    query: { authorId: req.query.userId, status: TicketStatus.open },
+    query: {
+      authorId: req.params.userId,
```
Comment (binaryoverload) [inline src/routes/tickets/controllers/getUserAuthoredTickets.ts:19]: Do you have input validation?

**Review by binaryoverload — CHANGES_REQUESTED** [review]
Much better with the base branch changed, thanks! :) 

Couple of questions...!

File: src/routes/tickets/controllers/getUserAuthoredTickets.ts (hunk 8 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/getUserAuthoredTickets.ts:19]: Very simple check if the userId exists. I should add better validation here.

File: src/routes/tickets/controllers/getUnassignedTickets.ts (hunk 7 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/getUnassignedTickets.ts]: Fixed.

File: src/routes/tickets/controllers/getUserAuthoredTickets.ts (hunk 8 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/getUserAuthoredTickets.ts]: Fixed.

File: src/routes/tickets/controllers/ticketController.ts [hunk 9]
```diff
@@ -82,6 +83,11 @@ const validate = (method: IValidationMethods['method']): RequestHandler[] => {
         body('subject', 'Field subject failed validation').exists().isObject().notEmpty(),
       ];
     }
+    case 'getUserAuthoredTickets': {
+      return [
+        body('userId', 'Field userId failed validation').exists().isString().escape().isLength({ min: 0, max: 100 }),
```
Comment (binaryoverload) [inline src/routes/tickets/controllers/ticketController.ts:88]: Shouldn't this be params?

**Review by binaryoverload — CHANGES_REQUESTED** [review]

File: src/routes/tickets/controllers/ticketController.ts (hunk 9 above)
Comment (TitusEntertainment) [reply src/routes/tickets/controllers/ticketController.ts]: I'll use the check fn instead.

bild
